sled-agent/src/nexus.rs

```diff
@@ -59,3 +63,35 @@ impl LazyNexusClient {
         ))
     }
 }
+
+type NexusRequestFut = dyn Future<Output = ()> + Send;
+type NexusRequest = Pin<Box<NexusRequestFut>>;
+
+/// A queue of futures which represent requests to Nexus.
+pub struct NexusRequestQueue {
+    tx: mpsc::UnboundedSender<NexusRequest>,
+    _worker: JoinHandle<()>,
+}
+
+impl NexusRequestQueue {
+    /// Creates a new request queue, along with a worker which executes
+    /// any incoming tasks.
+    pub fn new() -> Self {
+        let (tx, mut rx) = mpsc::unbounded_channel();
```

Comment:
Should this be bounded (if we keep it at all)? At least add a comment about why it's safe/reasonable to be unbounded.